src/debug/dwarf/line_numbers.cr

```diff
@@ -359,24 +359,30 @@ module Debug
       @current_sequence_matrix : Array(Row)?
 
       private def register_to_matrix(sequence, registers)
-        file = sequence.file_names[registers.file]
-        path = sequence.include_directories[file[1]]
-
-        row = Row.new(
-          registers.address,
-          registers.op_index,
-          path,
-          file[0],
-          registers.line.to_i,
-          registers.column.to_i,
-          registers.end_sequence
-        )
-
-        if rows = @current_sequence_matrix
-          rows << row
-        else
-          matrix << (rows = [row])
-          @current_sequence_matrix = rows
+        # checking is_stmt should be enough to avoid "non statement" operations
+        # some of which have confusing line number 0.
+        # but some operations within macros seem to be useful and marked as !is_stmt
+        # so attempt to include them also
+        if registers.is_stmt || (registers.line.to_i > 0 && registers.column.to_i > 0)
```

Comment:
I'd suggest using early return: 

```cr
return unless registers.is_stmt || (registers.line.to_i > 0 && registers.column.to_i > 0)
```

Reply:
Good idea.  There is one line that causes some concern though https://github.com/crystal-lang/crystal/pull/8499/commits/8f37b7d4773d72638aeec38aaef210ef21d7471d#diff-64764ab83ff3c47926623e383084ec34R388 maybe best to still run it, just in case?

Reply:
Right, so it seems it's better to just leave it as-is. This line IIUC is needed to finish the flow.